### ISSUE TITLE
Is there a way to resize the board and the gametrees?

### ISSUE THREAD

**Issue by HackYardo** [aki65/aki65.github.io#114]
Now we can zoom, pan, rotate the board and zoom, pan the gametree. However, it seems missing a important function to resize and replace the board and the gametree to the default ones. On desktop, we could double-click to do it. On mobile, how to do that?

**Comment by aki65** [comment]
Let's look at these defaults one by one:
1) **pan and zoom of board**: Here the default has unique properties: shows the full board, uses the whole screen width. So you would often want to return there, hence there should be an easy way to do it. And there is already: Just zoom out and make the board smaller than it originally was. Then the app will snap the board to its original position and size.
2) **rotation of board**: This is mainly used when loading a kifu which has switched coordinates. In that case you won't want to return to the default until the study of this kifu is finished. In other words: Switching back to the original rotation doesn't occur that often. Moreover it's pretty easy to do even without a special mechanism. Just at most two rotations and you are done.
3) **zoom of gametree**: Here the default size of the nodes is just my blind guess. So I don't assume that users will be interested in restoring exactly that size.
4) **pan of gametree**: Here it would be interesting to get back to the current node after panning around. But that's already easy since the app keeps the current node visible when it's changed. So, if you have lost it while panning, just tap the "<-" button, then the "->" button, and you see your current node again.

Summing up, I don't see any necessity to add further mechanisms here. Moreover, as you already mentioned, there is no standard intuitive way to "go back to defaults" on mobile. So, even if I would add something, users wouldn't know about it.

**Comment by HackYardo** [comment]
Well, u're definitely right, but if u'd like to talk a bit more, I have a idea about the looking of a gametree —— "big small branches":
- the more earlier or shorter a branch is, it'll be more bigger
- the more later or longer a branch is, it'll be mor e smaller

How do u think about this kind of gametree looking?

**Comment by aki65** [comment]
I assume you want it to look like a real tree, sounds interesting. But when I checked how to implement such a look, I found two severe drawbacks:
1) Because of the different sizes there will be many more distinct graphical elements, so drawing the tree will be much slower, therefore panning and zooming will be far too slow.
2) On the smaller branches the nodes would have to be smaller too (the lines alone have no real visual effect) and therefore they would be difficult to hit (in order to jump there).

So, I will stick with the uniform look (like all other apps do).

**Comment by HackYardo** [comment]
Oh, u're undoubtedly right, the more widgets, the more slow a app will be. BTW, will BadukAI support "**Landscape/portrait mode**" someday? If the users rotate their phones or pads, the app will switch its layout between the two modes autoly.

**Comment by aki65** [comment]
I have this on my todo list, but it's tricky: In landscape mode the status bar takes away much more space than in portrait mode. Moreover the settings menu is far too long for the screen in landscape mode. And the differences between 16:9 and 20:9 aspect ratio are much more severe in landscape mode than in portrait mode (where I can just make the histogram higher to compensate for the difference).

So, I am not sure that I can find the time to resolve all these issues, but I will try ...

**Comment by HackYardo** [comment]
Thank you. Perhaps I can understand it: `a small feature, a big dev time`. For the talks is away from the issue topic, I close it first.
